src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentRuntimeNodeWriter.cs

```diff
@@ -408,6 +416,18 @@ public override void WriteComponent(CodeRenderingContext context, ComponentInter
                 context.RenderNode(capture);
             }
 
+            if (hasRenderMode)
+            {
+                // _builder.SetRenderMode(__renderMode);
+                context.CodeWriter.Write(_scopeStack.BuilderVarName);
+                context.CodeWriter.Write(".");
+                context.CodeWriter.Write(ComponentsApi.RenderTreeBuilder.SetRenderMode);
+                context.CodeWriter.Write("(");
+                context.CodeWriter.Write(ComponentsApi.RenderTreeBuilder.RenderModeVariableName);
```

Comment:
Hmm, I'm not sure we need it, because we're assigning to a specific interface, so we can skip the type check as the compiler is already going to give us type mismatch errors.